trex-container-app/app/pyfiles/trexmac.py

```diff
@@ -1,51 +1,26 @@
-from datetime import datetime
-from dateutil import parser
-from dateutil.tz import tzutc
-from kubernetes import client, config, watch
+import json
+
+from kubernetes import client, config
 from kubernetes.client.rest import ApiException
 
 from logger import log
 
-
-def watch_cr(queue):
-    group = "examplecnf.openshift.io"
-    version = "v1"
-    namespace = os.environ.get("NAMESPACE", "example-cnf")
-    plural = "cnfappmacs"
-
-    config.load_incluster_config()
-    custom_api = client.CustomObjectsApi()
-    w = watch.Watch()
-    now = datetime.utcnow().replace(tzinfo=tzutc())
-    for event in w.stream(custom_api.list_namespaced_custom_object,
-                          group=group, version=version, plural=plural, namespace=namespace):
-        if event['type'] == 'ADDED':
-            meta = event['object']['metadata']
-            spec = event['object']['spec']
-            created = parser.parse(meta['creationTimestamp'])
-            if created > now:
-                queue.put(event['object'])
-
-def get_macs(spec):
+def get_dst_mac():
     macs = []
-    for resource in spec['resources']:
-        for device in resource['devices']:
-            macs.append(device['mac'])
-    log.info("macs list - %s" % ",".join(macs))
-    return macs
-
-def get_cnfappmac_cr_values():
-    group = "examplecnf.openshift.io"
-    version = "v1"
-    namespace = "example-cnf"
-    plural = "cnfappmacs"
 
     config.load_incluster_config()
-    custom_api = client.CustomObjectsApi()
+    v1 = client.CoreV1Api()
     try:
-        resp = custom_api.list_namespaced_custom_object(group, version, namespace, plural)
-        for item in resp['items']:
-            macs = get_macs(item['spec'])
-            return macs
+        response = v1.list_namespaced_pod(namespace="example-cnf", label_selector="example-cnf-type=cnf-app")
+        # there is only one pod that matches this label, so let's iterate the networks over it
+        networks = json.loads(response.items[0].metadata.annotations['k8s.v1.cni.cncf.io/networks'])
+
+        for network in networks:
+            log.info("get_dst_mac - MAC found", network["mac"])
```

Comment:
```suggestion
            log.info("get_dst_mac - MAC found: %s" % network.get("mac"))
```

Stack traces are displayed when not passing the right payload to log.info